### ISSUE TITLE
Missing constant not detected in array

### ISSUE THREAD

**Issue by muglug** [vimeo/psalm#837]
```php
class A {
  const B = 1;
  const C = [B];
}
```
Expected: Const B not found
Actual: no issue
